### DISCUSSION TITLE
Uncaught Error: Too many re-renders. React limits the number of renders to prevent an infinite loop.

### DISCUSSION THREAD

**Discussion by 11jl98** [frontendbr/forum#2238] [Perguntas e Respostas]
hello, I have the following error: Uncaught Error: Too many re-renders. React limits the number of renders to prevent an infinite loop.

this happens when i run this.

`const [dataPropertyTable, setPropertyTable] = useState([])

    const saveOrUpdateProperty = async () => {
        try {
            if (dataProperty.id_propriedade === "") {
                const data = await propertyService.save(dataProperty)
                getPropertyByIdCustomer(data.id_cliente)
                setProperty(data)
                alert("Propriedade salva")

            }
            else {
                await propertyService.update(dataProperty)
                getPropertyByIdCustomer(id_cliente)

                alert("Propriedade editada")
            }

        } catch (error) {
            console.log(error)
        }
    const getPropertyByIdCustomer =  async (idCustomer) => {
        try {
            console.log('esntrou')
            const data = await propertyService.getPropertyByIdCustomer(idCustomer)
            setPropertyTable(data)
        } catch (error) {
            alert(error)
        }
    }
`

**Comment by gabriel-hahn** [comment]
Fala João! Poderia colar todo o código do teu componente aqui? Assim fica mais fácil de identificar o problema 👌🏼 
Não achei o erro só com esses dois métodos que tu listou, talvez existe mais algum `useEffect` que está gerando um _loop_ ou algo nesse sentido.
